templates/confluence-init.properties.erb

```diff
@@ -32,7 +32,7 @@
 # specify your directory below (don't forget to remove the '#' in front)
 
 # confluence.home=c:/confluence/data
-<% if scope.lookupvar('confluence::data_dir') -%>
+<% if ! scope.lookupvar('confluence::data_dir').empty?  -%>
```

Comment:
A better fix would likely be to set the default to `undef` to begin with. See https://github.com/voxpupuli/puppet-confluence/blob/dc96ab3f0c0cb26bbda219ef90099f472f92faa1/manifests/init.pp#L20